### PR TITLE
SALTO-1155: Update jest version

### PR DESCRIPTION
**Pull request by ori-moisis**
- Update jest and jest types to newer version
- Addresses [this](https://github.com/advisories/GHSA-5fw9-fq32-wv5p) security advisory
- Minor fixes to tests that were exposed by the more strict version of jest

---

- Probably easier to review each commit on its own
- The new jest version found a couple of real bugs in our tests and code, those were fixed in previous PRs

---
_Release Notes_: 
_None_

### CODE REVIEW

**Comment by ori-moisis** [comment]
build fails because of a yarn bug that was only fixed in yarn v2 ([link](https://github.com/yarnpkg/yarn/issues/2629#issuecomment-753480348))
I really hope to find a workaround...

**Review by noamhammer — DISMISSED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -73,7 +73,9 @@ jobs:
 
       - run: yarn config set cache-folder /mnt/ramdisk/yarn-cache
 
-      - run: yarn
+        # Yarn has some random race-condition failures, trying multiple times seems to help
+        # when the source of the problem is yarn itself and not the packages
+      - run: yarn || yarn || yarn
```
Comment (netama) [inline .circleci/config.yml:78]: will we always need this, or only until everything is upgraded?

File: packages/e2e-credentials-store/.eslintrc.js [hunk 2]
```diff
@@ -19,10 +19,18 @@ const deepMerge = require('../../build_utils/deep_merge')
 module.exports = deepMerge(
   require('../../eslintrc.js'),
   {
+    overrides: [
```
Comment (netama) [inline packages/e2e-credentials-store/.eslintrc.js:22]: just out of curiosity, why was this change needed?

**Review by netama — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (ori-moisis) [reply .circleci/config.yml:78]: this is probably only relevant whenever we change any dependency (otherwise the first "yarn" would just do nothing an succeed, so the other two won't run)
I think we should leave it in though because I wouldn't want to re-discover this yarn bug every time we upgrade a package

File: packages/e2e-credentials-store/.eslintrc.js (hunk 2 above)
Comment (ori-moisis) [reply packages/e2e-credentials-store/.eslintrc.js:22]: because of the dynalite config file, because it is a .js file eslint wanted to lint it.
in the base eslint config we set a project for all files (that is probably a mistake, but I did not want to deal with that as part of this PR) but because this config .js file is not really a part of the project it made eslint fail.
This configuration override makes it so only .ts files will have the project attribute set.

btw - The project attribute is require only for lint rules that rely on types, I think we don't have any such rules enabled so it might be possible to get rid of this altogether, but again, didn't want to deal with that as part of this PR
